sql/src/main/java/io/crate/planner/operators/FetchOrEval.java

```diff
@@ -432,7 +432,10 @@ private ExecutionPlan planWithEvalProjection(PlannerContext plannerContext, Exec
 
     @Override
     public LogicalPlan tryOptimize(@Nullable LogicalPlan pushDown) {
-        if (pushDown instanceof Order) {
+        // If there is fetch involved then we cannot pushDown the Order
+        // since the outputs of the source plan of FetchOrEval contain the _fetchId
+        // and maybe not the columns we want to apply the ordering.
+        if (doFetch == false && pushDown instanceof Order) {
```

Comment:
If the columns are available, then this should still work. We could abort the optimization, if the Order can't be pushed down.